### PR TITLE
Fix _WSGIAdapter and TestClient WSGI compliance

### PR DESCRIPTION
**Pull request by pslacerda**
The missing `exc_info=None` parameter was causing me throuble when playing with middlewares and `TestClient`. This fix shoud be straightforward.

### CODE REVIEW

File: apistar/test.py [hunk 1]
```diff
@@ -77,7 +77,9 @@ def send(self, request, *args, **kwargs):
         """
         raw_kwargs = {}
 
-        def start_response(wsgi_status, wsgi_headers):
+        def start_response(wsgi_status, wsgi_headers, exc_info=None):
+            if exc_info:
+                raise exc_info[0].with_traceback(exc_info[1], exc_info[2])
```
Comment (tomchristie) [inline apistar/test.py]: Should this instead be `raise exc_info[0](exc_info[1]).with_traceback(exc_info[2])`?

Eg. see https://docs.python.org/3/library/exceptions.html#BaseException.with_traceback and https://docs.python.org/3.6/library/sys.html#sys.exc_info

Comment (tomchristie) [reply apistar/test.py]: Also we *might* want to consider a test case.

**Comment by pslacerda** [comment]
For sure I use TestClient to have an batch API that emulate many requests when the client did just one. But I didn't expected to test the test framework.

I make lots of sense however.

**Comment by pslacerda** [comment]
If you already reviewed it, sorry for this ammend. My linter isn't working correctly.

**Pull request closed without merging**